test/dd/test_dd_noise_functionality.cpp

```diff
@@ -90,44 +90,101 @@ TEST_F(DDNoiseFunctionalityTest, DetSimulateAdder4TrackAPD) {
       {"0011", 0.0242454336917}, {"1011", 0.0262779844799},
       {"0111", 0.0239296920989}, {"1111", 0.0110373166627}};
 
-  std::array<std::array<dd::SparsePVecStrKeys, 2>, 2> results{};
-  for (const auto useDensityMatrixType : {false, true}) {
-    for (const auto applyNoiseSequentially : {false, true}) {
-      auto dd = std::make_unique<DensityMatrixTestPackage>(qc.getNqubits());
-
-      auto rootEdge = dd->makeZeroDensityOperator(qc.getNqubits());
-      dd->incRef(rootEdge);
-
-      const auto noiseEffects = {dd::AmplitudeDamping, dd::PhaseFlip,
-                                 dd::Depolarization, dd::Identity};
-
-      auto deterministicNoiseFunctionality =
-          dd::DeterministicNoiseFunctionality(
-              dd, qc.getNqubits(), 0.01, 0.02, 0.02, 0.04, noiseEffects,
-              useDensityMatrixType, applyNoiseSequentially);
-
-      for (auto const& op : qc) {
-        dd->applyOperationToDensity(rootEdge, dd::getDD(op.get(), *dd),
-                                    useDensityMatrixType);
-        deterministicNoiseFunctionality.applyNoiseEffects(rootEdge, op);
-      }
+  const std::array<std::array<dd::SparsePVecStrKeys, 2>, 2> results{};
+  auto dd = std::make_unique<DensityMatrixTestPackage>(qc.getNqubits());
 
-      const auto m = rootEdge.getSparseProbabilityVectorStrKeys(0.001);
-      results[static_cast<std::size_t>(useDensityMatrixType)]
-             [static_cast<std::size_t>(applyNoiseSequentially)] = m;
-    }
+  auto rootEdge = dd->makeZeroDensityOperator(qc.getNqubits());
+  dd->incRef(rootEdge);
+
+  const auto noiseEffects = {dd::AmplitudeDamping, dd::PhaseFlip,
+                             dd::Depolarization, dd::Identity};
+
+  auto deterministicNoiseFunctionality = dd::DeterministicNoiseFunctionality(
+      dd, qc.getNqubits(), 0.01, 0.02, 0.02, 0.04, noiseEffects);
+
+  for (auto const& op : qc) {
+    dd->applyOperationToDensity(rootEdge, dd::getDD(op.get(), *dd));
+    deterministicNoiseFunctionality.applyNoiseEffects(rootEdge, op);
   }
+
+  const auto m = rootEdge.getSparseProbabilityVectorStrKeys(0.001);
+
   // Expect that all results are the same
   static constexpr fp TOLERANCE = 1e-10;
-  for (auto& result : results) {
-    for (auto& j : result) {
+  for (const auto& result : results) {
+    for (const auto& j : result) {
       for (const auto& [key, value] : j) {
         EXPECT_NEAR(value, reference.at(key), TOLERANCE);
       }
     }
   }
 }
 
+TEST_F(DDNoiseFunctionalityTest, testingMeasure) {
+  std::mt19937_64 mt{0}; // NOLINT(cert-msc51-cpp)
```

Comment:
No real need to define that here. Every QuantumComputation comes with a properly seeded RNG that you can get via the `getGenerator` method (https://github.com/cda-tum/mqt-core/blob/fc7d8660e6d2007befbcdf70abdad0e845e67ae8/include/mqt-core/QuantumComputation.hpp#L314).

Reply:
I changed the code as suggested.